src/Request.ts

```diff
@@ -18,6 +16,9 @@ type ResponseTypeMap = {
   json: JsonValue;
 };
 
+// The type system allows us infer the response type from the requested `responseType`
+type ResponseType = keyof ResponseTypeMap;
```

Comment:
DX-wise I'd keep it the original way:

<img width="509" alt="Screenshot 2024-03-11 at 11 22 40" src="https://github.com/swan-io/request/assets/1688645/80f52d74-4869-4b87-847d-f9d69127f4e2">

<img width="506" alt="Screenshot 2024-03-11 at 11 22 49" src="https://github.com/swan-io/request/assets/1688645/cf32a2bd-6428-40f8-8ef9-b53a0da5990b">